docs/MarketingCloud-batchsource.md

```diff
@@ -0,0 +1,96 @@
+# Salesforce Marketing Cloud Source
+
+Description
+-----------
+
+Reads from one or multiple objects withing Salesforce Marketing Cloud depending on the mode values set for this plugin. 
+In case of `Multi Object` mode, the source will output a record for each row from the object it reads, with each record
+containing an additional field that holds the name of the object from which the record came from. In case of 
+`Single Object` mode, this additional field will not be there in the output. In addition, for each object that will be 
+read, this plugin will set pipeline arguments where the key is `multisink.[objectname]` and the value is the schema of 
+the object.
+
+Properties
+----------
+
+**Reference Name**: Name used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**Data Retrieval Mode**: Mode of data retrieval. The mode can be one of two values: 
+
+`Multi Object` - will allow user to fetch data for multiple data extensions,  
+
+`Single Object` - will allow user to fetch data for single data extension.
+
+**Object**: Specify the object for which data to be fetched. This can be one of following values: 
+
+`Data Extension` - will allow user to fetch data for a single Data Extension object,
+
+`Email` - will allow user to fetch data for Email object,
+
+`Mailing List` - will allow user to fetch data for Mailing List object,
+
+`Bounce Event` - will allow user to fetch Tracking Bounce Events,
+
+`Open Event` - will allow user to fetch Tracking Open Events,
+
+`Unsub Event` - will allow user to fetch Tracking UnSubscribe Events,
+
+`Sent Event` - will allow user to fetch Tracking Sent Events,
+
+`UnSent Event` - will allow user to fetch Tracking UnSent Events.
+
+Note, this value will be ignored if the Mode is set to `Multi Object`.  
+
+**Data Extension External Key**: Specify the data extension key from which data to be fetched. Note, this value will 
+be ignored in following two cases: 
+
+* If the Mode is set to `Multi Object`
+* If the selected object name is other than `Data Extension`. 
+
+**Object List**: Specify the comma-separated list of objects for which data to be fetched; for example: 
+'Object1,Object2'. This can be one or more values from following possible values: 
+
+`Data Extension` - will allow user to fetch data for a single Data Extension object,
+
+`Email` - will allow user to fetch data for Email object,
+
+`Mailing List` - will allow user to fetch data for Mailing List object
+
+`Bounce Event` - will allow user to fetch Tracking Bounce Events,
+
+`Open Event` - will allow user to fetch Tracking Open Events,
+
+`Unsub Event` - will allow user to fetch Tracking UnSubscribe Events,
+
+`Sent Event` - will allow user to fetch Tracking Sent Events,
+
+`UnSent Event` - will allow user to fetch Tracking UnSent Events.
```

Comment:
UnSent is not English it’s programming. Could we define this as normal English words please.

Reply:
Done